scripts/sanity_chk/sanitylib.py

```diff
@@ -581,7 +582,15 @@ def handle(self):
             elif runner == "jlink":
                 command.append("--tool-opt=-SelectEmuBySN  %s" % (board_id))
 
-        serial_device = hardware['serial']
+        serial_pty = hardware['serial_pty']
+        if serial_pty:
+            master, slave = pty.openpty()
+            ser_pty_process = subprocess.Popen(serial_pty, stdout=master, stdin=master, stderr=master)
+            serial_device = os.ttyname(slave)
+        else:
+            serial_device = hardware['serial']
+
+        print("Using serial device ", serial_device)
```

Comment:
Space at the end of the string is redundant, that's because default separator is space. Therefore, there will be two spaces after "device" word.

Reply:
Fixed